cypress/e2e/item/shortLink/shortLink.cy.ts

```diff
@@ -0,0 +1,418 @@
+      });
+
+      it('Add default short link', () => {
+        cy.visit(buildItemPath(itemId));
+        cy.get(`#${buildShareButtonId(itemId)}`).click();
+
+        cy.wait('@getShortLinksItem').then(
+          ({ response: { body: shortLinksItemRes } }) => {
+            expect(shortLinksItemRes.length).equals(0);
+
+            cy.get(`.${SHORT_LINK_COMPONENT}`).should('have.length', 0);
+            cy.get(`#${SHORT_LINK_NEW_BUTTON_ID}`).click();
+
+            cy.wait('@checkShortLink').then(
+              ({ response: { statusCode: checkStatusCode } }) => {
+                expect(checkStatusCode).equals(StatusCodes.NOT_FOUND);
+
+                cy.get(`#${SHORT_LINK_SAVE_BUTTON_ID}`).click();
+
+                cy.wait('@postShortLink').then(
+                  ({ response: { statusCode: postCode, body: postBody } }) => {
+                    expect(postCode).equals(StatusCodes.OK);
+                    expect(postBody.platform).equals(Context.Player);
+
+                    cy.wait('@getShortLinksItem').then(
+                      ({ response: { body } }) => {
+                        expect(body.length).equals(1);
+                      },
+                    );
+                  },
+                );
+              },
+            );
+          },
+        );
+      });
+
+      it('Add a custom short link', () => {
+        const SHORT_LINK_ALIAS = 'test-1';
+        const SHORT_LINK_PLATFORM = Context.Builder;
+
+        cy.visit(buildItemPath(itemId));
+        cy.get(`#${buildShareButtonId(itemId)}`).click();
+
+        cy.wait('@getShortLinksItem').then(
+          ({ response: { body: shortLinksItemRes } }) => {
+            expect(shortLinksItemRes.length).equals(0);
+
+            cy.get(`.${SHORT_LINK_COMPONENT}`).should('have.length', 0);
+            cy.get(`#${SHORT_LINK_NEW_BUTTON_ID}`).click();
+
+            cy.wait('@checkShortLink').then(
+              ({ response: { statusCode: checkStatusCode } }) => {
+                expect(checkStatusCode).equals(StatusCodes.NOT_FOUND);
+
+                cy.get(`#${SHORT_LINK_ALIAS_INPUT_ID}`)
+                  .clear()
+                  .type(SHORT_LINK_ALIAS);
+
+                cy.get(`#${SHORT_LINK_PLATFORM_SELECT_ID}`).click();
+                cy.get(`li[data-value="${SHORT_LINK_PLATFORM}"]`).click();
+                cy.get(`#${SHORT_LINK_SAVE_BUTTON_ID}`).click();
+
+                cy.wait('@postShortLink').then(
+                  ({ response: { statusCode: postCode, body: postBody } }) => {
+                    expect(postCode).equals(StatusCodes.OK);
+                    expect(postBody.platform).equals(SHORT_LINK_PLATFORM);
+                    expect(postBody.alias).equals(SHORT_LINK_ALIAS);
+
+                    cy.wait('@getShortLinksItem').then(
+                      ({ response: { body } }) => {
+                        expect(body.length).equals(1);
+                      },
+                    );
+                  },
+                );
```

Comment:
I don't think you need to nest it that much. Plus if you only have expect in the middle you can wait for many calls in the same `cy.wait([@,@,@])` I think.